### PR TITLE
Disable Validation API if Analysis is disabled

### PR DESCRIPTION
**Pull request by crisptrutski**
### Description

The query validation tool will not behave correctly if query analysis is disabled.

This PR gates the API behind that feature, and makes it visible to admins.

Thread here: https://metaboat.slack.com/archives/C06V3JMECH2/p1721809657496249

### CODE REVIEW

File: enterprise/backend/src/metabase_enterprise/query_reference_validation/api.clj [hunk 1]
```diff
@@ -89,4 +90,14 @@
      :limit  mw.offset-paging/*limit*
      :offset mw.offset-paging/*offset*}))
 
-(api/define-routes api/+check-superuser +auth)
+(defn +check-setting
+  "Middleware that returns a 401 response if `request` has no associated `:metabase-user-id`."
```
Comment (crisptrutski) [inline enterprise/backend/src/metabase_enterprise/query_reference_validation/api.clj:94]:
```suggestion
  "Middleware that gates this API behind the associated feature flag"
```

**Review by tsmacdonald — APPROVED** [review]
Neat! Is @npfitz aware that this is a potential response?

**Comment by crisptrutski** [comment]
@tsmacdonald Yes, thread [here](https://metaboat.slack.com/archives/C06V3JMECH2/p1721809657496249)

**Review by npfitz — APPROVED** [review]
🔥

**Pull request merged**